src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewList.cs

```diff
@@ -0,0 +1,689 @@
+						m_emptySlotIndex = size - 1;
+					}
+					else if (belowIndex > 0 && m_draggedTreeViewNode != null)
+					{
+						m_emptySlotIndex = -1;
+						TreeViewNode tvi = m_draggedTreeViewNode;
+						if (ListViewModel.IndexOfNode(tvi, out var draggedIndex))
+						{
+							int indexToUse = (draggedIndex < belowIndex) ? belowIndex : (belowIndex - 1);
+							var item = ContainerFromIndex(indexToUse);
+							if (item != null)
+							{
+								var treeViewItem = (TreeViewItem)item;
+								var pointInsideItem = args.GetPosition((UIElement)treeViewItem);
+
+								// if the point is in the top half of the item
+								// we need to insert before that item
+								if (pointInsideItem.Y < treeViewItem.ActualHeight / 2)
+								{
+									m_emptySlotIndex = belowIndex - 1;
+								}
+								else
+								{
+									m_emptySlotIndex = belowIndex;
+								}
+							}
+						}
+					}
+					else
+					{
+						// top of the list
+						m_emptySlotIndex = 0;
+					}
+
+					bool allowReorder = true;
+					if (IsFlatIndexValid(m_emptySlotIndex))
+					{
+						var insertAtNode = NodeAtFlatIndex(m_emptySlotIndex);
+						if (IsMultiselect)
+						{
+							// If insertAtNode is in the selected items, then we do not want to allow a dropping.
+							var selectedItems = ListViewModel.SelectedNodes;
+							for (var i = 0; i < selectedItems.Count; i++)
+							{
+								var selectedNode = selectedItems[i];
+								if (selectedNode == insertAtNode)
+								{
+									allowReorder = false;
+									break;
+								}
+							}
+						}
+						else
+						{
+							if (m_draggedTreeViewNode != null && insertAtNode != null && insertAtNode.Parent != null)
+							{
+								var insertContainer = ContainerFromNode(insertAtNode.Parent);
+								if (insertContainer != null)
+								{
+									allowReorder = ((TreeViewItem)insertContainer).AllowDrop;
+								}
+							}
+						}
+					}
+					else
+					{
+						// m_emptySlotIndex does not exist in the ViewModel - don't allow the reorder.
+						allowReorder = false;
+					}
+
+					if (allowReorder)
+					{
+						args.AcceptedOperation = DataPackageOperation.Move;
+					}
+					else
+					{
+						m_emptySlotIndex = -1;
+						args.AcceptedOperation = DataPackageOperation.None;
+						args.Handled = true;
+					}
+				}
+
+				UpdateDropTargetDropEffect(false, false, null);
+			}
+			base.OnDragOver(args);
+		}
+
+		protected override void OnDragEnter(DragEventArgs args)
+		{
+			if (!args.Handled)
+			{
+				UpdateDropTargetDropEffect(false, false, null);
+			}
+			base.OnDragEnter(args);
+		}
+
+		protected override void OnDragLeave(DragEventArgs args)
+		{
+			m_emptySlotIndex = -1;
+			base.OnDragLeave(args);
+			if (!args.Handled)
+			{
+				UpdateDropTargetDropEffect(false, true, null);
+			}
+		}
+
+		protected override void PrepareContainerForItemOverride(DependencyObject element, object item)
+		{
+			var itemNode = (TreeViewNode)NodeFromContainer(element));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)